tests/pretty/autodiff_reverse.rs

```diff
@@ -0,0 +1,13 @@
+#![feature(autodiff)]
+//@ pretty-mode:expanded
+//@ pretty-compare-only
+//@ pp-exact:autodiff_reverse.pp
+
+// Test that reverse mode ad macros are expanded correctly.
+
+#[autodiff(df, Reverse, Duplicated, Const, Active)]
```

Comment:
Problem && Suggestion: I think we want at least test cases for:

- Positive test cases for what `autodiff` *should* be applied to.
- Negative test cases for when `autodiff` is misapplied to AST nodes and their diagnostics.
- `#[autodiff]` malformed attribute (where args?) error
- `#[autodiff = ""]` invalid attribute syntax
- `#[autodiff()]` where arg
- `#[autodiff(df)]` + `fn df()` <- what if I already have a `df` in value namespace?
- `#[autodiff(df, Reverse)]` + `enum Foo { Reverse }` + `use Foo::Reverse;` <- what if I already have a `Reverse` in type (enum variant decl) and value (enum variant constructor) namespace?
- `#[autodiff(df)]` <- is this minimally acceptable?
- `#[autodiff(df, Reverse)]` <- valid mode
- `#[autodiff(df, Debug)]` <- invalid mode
- `#[autodiff(df, Forward, Reverse)]` <- is this valid
- target fn has specified valid return type e.g. `-> f64`
- target fn has unspecified return type `fn foo() {}`
- target fn has specified but invalid return type e.g. `-> Owo`
- target fn has aliased f32/f64 return types (currently unsupported) `-> F64Alias`
- target fn has `#[repr(transparent)] struct F64Trans { inner: f64 }` return type (currently unsupported) `-> F64Trans`.

Reply:
All of these cases now give error messages instead of ICEs.
I also made sure that we don't return after the first error, but continue to do something sufficiently okish that we can first expand all autodiff macros before we abort compilation.
We probably could include even better errors in the future, but I left those as fixme's for now. 
Do they look good to you?

Reply:
Yep, for an initial set of "surface-level" error tests, they look good to me.